database-migration/src/test/java/uk/ac/ebi/eva/dbmigration/mongodb/ExtractAnnotationFromVariantTest.java

```diff
@@ -0,0 +1,302 @@
+import static uk.ac.ebi.eva.dbmigration.mongodb.ExtractAnnotationFromVariant.XREF_ID_FIELD;
+
+/**
+ * Test {@link ExtractAnnotationFromVariant}
+ */
+public class ExtractAnnotationFromVariantTest {
+
+    private static final String VEP_VERSION = "88";
+
+    private static final String CACHE_VERSION = "90";
+
+    private static final String VARIANT_COLLECTION_NAME = "variantsCollection";
+
+    private static final String ANNOTATION_COLLECTION_NAME = "annotationCollection";
+
+    private static final String ANNOTATION_METADATA_COLLECTION_NAME = "annotationMetadataCollection";
+
+    private static final String READ_PREFERENCE = "primary";
+
+    private ExtractAnnotationFromVariant extractAnnotationFromVariant;
+
+    @Before
+    public void setUp() throws Exception {
+        extractAnnotationFromVariant = new ExtractAnnotationFromVariant();
+    }
+
+    @Test
+    public void variantWithoutAnnotationShouldNotChange() {
+        String dbName = "variantWithoutAnnotation";
+
+        Properties properties = new Properties();
+        properties.put(DatabaseParameters.VEP_VERSION, VEP_VERSION);
+        properties.put(DatabaseParameters.VEP_CACHE_VERSION, CACHE_VERSION);
+        properties.put(DatabaseParameters.DB_NAME, dbName);
+        properties.put(DatabaseParameters.DB_COLLECTIONS_VARIANTS_NAME, VARIANT_COLLECTION_NAME);
+        properties.put(DatabaseParameters.DB_COLLECTIONS_ANNOTATIONS_NAME, ANNOTATION_COLLECTION_NAME);
+        properties.put(DatabaseParameters.DB_COLLECTIONS_ANNOTATION_METADATA_NAME, ANNOTATION_METADATA_COLLECTION_NAME);
+        properties.put(DatabaseParameters.DB_READ_PREFERENCE, READ_PREFERENCE);
+        DatabaseParameters databaseParameters = new DatabaseParameters();
+        databaseParameters.load(properties);
+        ExtractAnnotationFromVariant.setDatabaseParameters(databaseParameters);
+
+        MongoDatabase database = new Fongo("testServer").getMongo().getDatabase(dbName);
+        MongoCollection<Document> variantsCollection = database.getCollection(VARIANT_COLLECTION_NAME);
+        MongoCollection<Document> annotationCollection = database.getCollection(ANNOTATION_COLLECTION_NAME);
+
+
+        Document variantWithoutAnnot = Document.parse(VariantData.VARIANT_WITHOUT_ANNOT);
+        variantsCollection.insertOne(variantWithoutAnnot);
+
+        extractAnnotationFromVariant.migrateAnnotation(database);
+        extractAnnotationFromVariant.reduceAnnotationFromVariants(database);
+
+        try (MongoCursor<Document> variantCursor = variantsCollection.find().iterator()) {
+            while (variantCursor.hasNext()) {
+                Document variantObj = variantCursor.next();
+                assertNull(variantObj.get(ANNOT_FIELD));
+            }
+        }
+
+        assertEquals(0, variantsCollection.count(new BasicDBObject(ANNOT_FIELD, new BasicDBObject("$exists", true))));
+        assertEquals(0, annotationCollection.count());
```

Comment:
the annotation metadata changeSet does not know whether there actually was annotation objects  migrated or not, if adds VEP